.github/workflows/sync-upstream-change.yml

```
@@ -0,0 +1,72 @@
+name: Sync with Upstream (change)
+
+on:
+  pull_request_review:
+    types: [edited, submitted]
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    if: startsWith(github.event.pull_request.head.ref, 'sync-upstream-') && github.event.review.state == 'changes_requested' && github.event.review.body == 'bot change'
```

nit: line is a bit long

```suggestion
    if: >
      startsWith(github.event.pull_request.head.ref, 'sync-upstream-') &&
      github.event.review.state == 'changes_requested' &&
      github.event.review.body == 'bot change'
```